src/tests/wrapMeInValidComponents.js

```diff
@@ -0,0 +1,12 @@
+import { Provider } from 'react-redux';
+import { BrowserRouter } from 'react-router-dom';
+import store from '../Redux/store';
+
+const wrapper = (ele) => (
+  <BrowserRouter>
+    <Provider store={store}>{ele}</Provider>
+    ;
+  </BrowserRouter>
+);
+
+export default wrapper;
```

Comment:
Be aware that in order to ensure no dependencies between tests, each one of them should have an instance of the store and not the main one itself as stated in the documentation:

> Additionally, the test code should create a separate Redux store instance for every test, rather than reusing the same store instance and resetting its state. That ensures no values accidentally leak between tests.

To solve this, we can use the [solution](https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function) provided by redux to create a wrapper with similar functionality.

Reply:
- Actually, these concepts are the only reason holding me from submitting the card till now, But considering the time and the submission date, I was thinking we can do more test on latter time.

Reply:
As you wish, this change has to be done as part of the setup for almost all tests. We can get together and see what are the priorities for now.

Reply:
Also, I don't know if you notice buddy, but there are tests With react test library, It's testing pretty much basic. but it exists.
# 😊